type_lens/function_view.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+import inspect
+import sys
+from typing import TYPE_CHECKING, Any, Callable
+
+from typing_extensions import get_type_hints
+
+from type_lens.parameter_view import ParameterView
+from type_lens.type_view import TypeView
+
+__all__ = ("FunctionView",)
+
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+
+class FunctionView:
```

Comment:
How about `CallableView`? I.e., should this handle methods, and class instances with `__call__()` defined, etc?

Reply:
`SignatureView`?

Reply:
I think between `SignatureView` and `CallableView`, i prefer callable. to me the `@classmethod` constructor below would be the indicator of the name of the type.

the fact that the `__call__` of each of these things is a function:

```python
>>> print.__class__
<class 'builtin_function_or_method'>

>>> # or

>>> class Foo:
...    def __init__(self): ...
>>> Foo.__init__.__class__
<class 'function'>
```

What i dont like about `callable` is that it's a protocol, not a concrete type/thing, whereas `Type(View)` and `Parameter(View)` **are** concrete things. Thus i still personally think Function.

**BUT**

> something implementing the `__call__` method

the "thing" in this case is some `Foo` class or class instance, not the actual init/call method. So in that regard it really **does** accept a callable, and then return the annotations of the corresponding function. but it's not like you're sending `Foo.__init__`. Thus I'd be happy with callable too.

---

Fun fact, while you get a function if you define your own init method. if you dont:

```python
>>> class Foo: ...
...
>>> Foo.__init__
<slot wrapper '__init__' of 'object' objects>
>>> Foo.__init__.__class__
<class 'wrapper_descriptor'>
>>> Foo.__init__.__class__.__mro__
(<class 'wrapper_descriptor'>, <class 'object'>)
>>> Foo.__init__.__class__.__call__
<slot wrapper '__call__' of 'wrapper_descriptor' objects>
```

Which doesn't appear to have any relation to function.